CoreScriptsRoot/Modules/Settings/Pages/ReportAbuseMenu.lua

```diff
@@ -190,39 +229,41 @@ local function Initialize()
 		end
 
 		local function onReportSubmitted()
-			local abuseReason = nil
-			if this.GameOrPlayerMode.CurrentIndex == 2 then
-				abuseReason = ABUSE_TYPES_PLAYER[this.TypeOfAbuseMode.CurrentIndex]
-
-				local currentAbusingPlayer = this:GetPlayerFromIndex(this.WhichPlayerMode.CurrentIndex)
-				if currentAbusingPlayer and abuseReason then
-					spawn(function()
-						game.Players:ReportAbuse(currentAbusingPlayer, abuseReason, this.AbuseDescription.Selection.Text)
-					end)
-				end
-			else
-				abuseReason = ABUSE_TYPES_GAME[this.TypeOfAbuseMode.CurrentIndex]
-				if abuseReason then
-					spawn(function()
-						game.Players:ReportAbuse(nil, abuseReason, this.AbuseDescription.Selection.Text)
-					end)
+			if areFieldsCompleted() then
```

Comment:
lets switch this to the negative comparison and just return immediately, makes the code a little less messy